File: 75-sort-colors/75-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        return solve(nums);
    }
private:
    void solve(vector<int>& nums) {
        int low = 0, i = 0, high = nums.size() - 1;
        
        while(low <= high) {
            if(nums[low] == 0) {
                swap(nums[low], nums[i]);
                ++i;
                ++low;
            }
            else if(nums[low] == 2) {
                swap(nums[low], nums[high]);
                --high;
            }
            else
                ++low;
        }
    }
};

File: 75-sort-colors/75-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        n = len(nums)
        start, mid, end = 0, 0, n - 1
        
        while start <= end:      
            if nums[start] == 0:
                nums[start], nums[mid] = nums[mid], nums[start]
                mid += 1
                start += 1
                
            elif nums[start] == 2:
                nums[start], nums[end] = nums[end], nums[start]
                end -= 1
                
            else:
                start += 1
        